File: packages/core/tests/fixtures/javascript/code/classes/basic_class.js
/**
 * ES6 class definitions
 * Tests: class declarations, constructors, methods
 */

class User {
  constructor(name, email) {
    this.name = name;
    this.email = email;
    this.active = true;
  }

  greet() {
    return `Hello, ${this.name}!`;
  }

  getInfo() {
    return {
      name: this.name,
      email: this.email,
      active: this.active,
    };
  }

  activate() {
    this.active = true;
  }

  deactivate() {
    this.active = false;
  }
}

class Animal {
  constructor(name) {
    this.name = name;
  }

  speak() {
    throw new Error("Subclass must implement speak()");
  }
}

class Dog extends Animal {
  static getSpecies() {
    return "Canis familiaris";
  }

  speak() {
    return "Woof!";
  }

  fetch(item) {
    return `${this.name} fetched ${item}`;
  }
}

// Example usage demonstrating constructor calls, method calls, and static methods
function main() {
  const myDog = new Dog("Buddy");
  myDog.speak();
  Dog.getSpecies();
  console.log(myDog.fetch("ball"));
}

module.exports = { User, Animal, Dog, main };

File: packages/core/tests/fixtures/python/code/classes/basic_class.py
"""
Basic class definition
Tests: class, __init__, instance methods
"""

class User:
    def __init__(self, name: str, email: str):
        self.name = name
        self.email = email
        self.active = True

    def activate(self):
        self.active = True

    def deactivate(self):
        self.active = False

    def get_info(self):
        return f"{self.name} ({self.email})"


class Product:
    def __init__(self, id: int, name: str, price: float):
        self.id = id
        self.name = name
        self.price = price

    def apply_discount(self, discount: float):
        self.price = self.price * (1 - discount)

    def get_price(self):
        return self.price
